### PR TITLE
Valentines day improvements nobody asked

### PR DESCRIPTION
**Pull request by PowerfulBacon**
*

## About The Pull Request

Does some improvements to valentines day:
 - When the event starts you get 5 minutes to send a letter
 - You can send the letter to anyone on the same z-level as you (In the first 5 minutes nukies can date other nukies)
 - You now gain the factions that your lover has
 - You only get 1 shot to choose someone to send it to to prevent metaing the list of names to find potential nukies or other non station dwellers.
 - If the person you sent your card to sends a card back you automatically get each other marked as your protect target.
 - After 5 minutes if you don't have a protect target, you will be assigned a random one. This is wear AIs and non humans get assigned.

## Why It's Good For The Game

Improves interaction with certain antagonists, allows you to choose better protect targets that won't just ignore their objective the entire round or go get themselves killed (obviously unless you get rejected lol).

Also much like in real life, the second valentines day starts you have 5 minutes before you are considered a failure and get automatically put on a date with a completely random person in the world you have never met before (Or an AI.)

:cl:Totally not Bacon
Add: Improves valentines day, gives ai proper valentines laws.
tweak: You can now send a card to someone by writing their name on it with a pen
/:cl:

### CODE REVIEW

**Comment by Sinestia** [comment]
10/10 will fix my crippling sorrow and isolation

||fr this is actually cute do it||

**Comment by sircope** [comment]
metagangs: the gamemode

**Comment by PowerfulBacon** [comment]
should probably stop people planting C4s on the cards and then sending them to people lol

File: code/game/objects/items/grenades/plastic.dm [hunk 1]
```diff
@@ -105,6 +105,11 @@
 		return
 	if(ismob(AM) && !can_attach_mob)
 		return
+	if(istype(AM, /obj/item/valentine))
```
Comment (ike709) [inline code/game/objects/items/grenades/plastic.dm]: There's seriously no better way to do this? Preferably on the valentine item?

**Review by utilbird — DISMISSED** [review]
Syndicate glue should probably be blocked lmao

**Comment by OceanFish1** [comment]
Erp???? Also metagang

**Comment by ToasterBan** [comment]
will you be my valentine 👉👈

**Comment by super12pl** [comment]
Thats terrible idea
please merge it

**Comment by ghost** [comment]
This is a horrible idea.
do it no balls

File: code/game/objects/items/grenades/plastic.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/plastic.dm]: No, I thought there would be some c4act proc, but on planting the c4 just moves to null space and then blows the item up after x seconds

**Comment by ghost** [comment]
How about putting the card next to SM for a while before sending?

**Comment by PowerfulBacon** [comment]
ok you know what im just going to make a new card when you send it

File: code/game/objects/items/grenades/plastic.dm (hunk 1 above)
Comment (bloons3) [reply code/game/objects/items/grenades/plastic.dm]: Is this still required with creating the new valentine card?

**Comment by ZombiesVsPlants** [comment]
aww ai doesn't get to choose a valentine

**Comment by ZombiesVsPlants** [comment]
also, will the protect objective be above or below ai laws?

**Comment by PowerfulBacon** [comment]
Protecting date comes above all loyalties however with AI it starts to get funky. Ill make a hacked law above all others

**Comment by ZombiesVsPlants** [comment]
eyy based

File: code/modules/events/holiday/vday.dm [hunk 2]
```diff
@@ -59,23 +69,80 @@
...
+			//wow.
+			forge_valentines_objective(sender, target)
+		//Off it goes!
+		//Create a new card to prevent exploiting
+		var/obj/item/valentine/new_card = new(get_turf(picked_human))
+		new_card.message = message
+		new_card.sender = user
+		new_card.target = picked_human
+		new_card.name = "valentines card from [new_card.sender]"
+		new_card.desc = "A Valentine's card! It is addressed to [new_card.target]."
```
Comment (utilbird) [inline code/modules/events/holiday/vday.dm:142]: Should the new card have `used` set to true?

Comment (PowerfulBacon) [reply code/modules/events/holiday/vday.dm:142]: oh I thought I set that

Comment (PowerfulBacon) [inline code/modules/events/holiday/vday.dm:142]:
```suggestion
		new_card.desc = "A Valentine's card! It is addressed to [new_card.target]."
		new_card.used = TRUE
```

**Comment by DatBoiTim** [comment]
>AI gets hacked law

Hmmmmmm yes I see this going well

**Comment by PowerfulBacon** [comment]
> 
> 
> > AI gets hacked law
> 
> Hmmmmmm yes I see this going well

It was allowed to bypass its laws anyway before

File: code/game/objects/items/grenades/plastic.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/plastic.dm]: no

File: code/game/objects/items/grenades/plastic.dm [hunk 3]
```diff
@@ -105,6 +105,11 @@
...
+		to_chat(user, "<span class='userdanger'>Really...?</span>")
+		to_chat(user, "<span class='warning'>You start feeling incredibly depressed and horrified at what you tried to do.</span>")
+		ADD_TRAIT(user, TRAIT_DEPRESSION, ROUNDSTART_TRAIT)
+		return
```
Comment (PowerfulBacon) [inline code/game/objects/items/grenades/plastic.dm]:
```suggestion
```

**Review by francinum — APPROVED** [review]

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**